src/drivers/intel/cavs/dmic.c

```diff
@@ -239,23 +243,28 @@ static uint64_t dmic_work(void *data, uint64_t delay)
 					 CIC_CONTROL_MIC_MUTE_BIT, 0);
 
 		if (dmic->startcount == DMIC_UNMUTE_FIR) {
-			if (dmic->fifo_a)
+			switch (dai->index) {
+			case 0:
 				dmic_update_bits(dai, base[i] + FIR_CONTROL_A,
 						 FIR_CONTROL_A_MUTE_BIT, 0);
-
-			if (dmic->fifo_b)
+				break;
+			case 1:
 				dmic_update_bits(dai, base[i] + FIR_CONTROL_B,
 						 FIR_CONTROL_B_MUTE_BIT, 0);
+				break;
```

Comment:
@singalsu no default case here?

Reply:
The index has been checked earlier in dmic_set_config() so the check would be redundant. There's quite many similar parts in code so tracing an error in every usage would add code size. Does our coding style require such?